webrender/res/prim_shared.glsl

```diff
@@ -758,4 +758,13 @@ float do_clip() {
     return vClipMaskUvBounds.xy == vClipMaskUvBounds.zw ? 1.0:
         all(inside) ? textureLod(sCacheA8, vClipMaskUv, 0.0).r : 0.0;
 }
+
+vec4 dither(vec4 color) {
+    const int matrix_mask = 7;
+
+    ivec2 pos = ivec2(gl_FragCoord) & ivec2(matrix_mask);
```

Comment:
I think it would be safer to use `gl_FragCoord.xy` here, given that without it you'd be asking for 2 conversions done: float to int and vec4 to vec2, which some GLSL compilers may not appreciate.